scrapper.py

```diff
@@ -26,38 +39,140 @@ class Crawler:
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        articles = article_bs.find('div', {'class': 'entry-content'})
+        all_links = articles.find_all('a')[1:]
+        for link in all_links:
+            if len(self.urls) < self.max_articles:
+                if re.match(r'https?://vestnik\.lunn\.ru/', link['href']):
+                    self.urls.append(link['href'])
+                else:
+                    self.urls.append(DOMAIN_NAME + link['href'])
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for url in self.seed_urls:
+            response = requests.get(url, headers=HEADERS)
+            sleep(4)
+
+            if not response.ok:
+                break
```

Comment:
weak error handling. OK, we have stoped the loop - have we collected enough URLs?